### ISSUE TITLE
does resolveIds support parallel evaluation of asyncIds

### ISSUE THREAD

**Issue by RamprasadAkella** [doowb/async-helpers#17]
I observed if I call resolveIds with a string having multiple async ids in it, it is evaluating all the async ids in series. Does it support parallel evaluation of ids ?

**Comment by doowb** [comment]
tldr; No, it doesn't.

I think that would lead to some weird issues when resolving ids that may have been passed into other helper functions as arguments. Each id is only resolved once and the value is cached in case it's called again. If we did parallel, then an id might be resolving when another method also tries to resolve it, then the helper might be executed multiple times. I'm sure there are ways to ensure that won't happen but I think it would be a more complicated code.

I'll probably be refactoring this code soon to make it more maintainable. Once that is done, we could look at parallel execution, but I don't think it's a good fit at this time.
